docs/rendering_system.rst

```diff
@@ -0,0 +1,241 @@
+
+
+2. Orchestration
+----------------
+
+``build.py`` acts to manage the rendering process.
+Usage is covered in :doc:`build`.
+
+
+3. Extension
+------------
+
+The Sphinx extension and theme are contained in the ``pep_sphinx_extensions``
+directory.
+The following is a brief overview of the stages of the PEP rendering process,
+and how the extension functions at each point.
+
+
+3.1 Extension setup
+~~~~~~~~~~~~~~~~~~~
+
+The extension registers several objects:
+
+- ``FileBuilder`` and ``DirectoryBuilder`` run the build process for file- and
+  directory- based building
+- ``PEPParser`` registers the custom document transforms and parses PEPs to
+  a Docutils document.
+- ``PEPTranslator`` converts a Docutils document into HTML
+- ``PEPRole`` handles ``:pep:`` roles in the reStructuredText source
+
+It also patches default behaviour:
+
+- updating the default settings
+- updating the Docutils inliner
+- using HTML maths display over MathJax
+
+
+3.2 Builder initialised
+~~~~~~~~~~~~~~~~~~~~~~~
+
+After the Sphinx builder object is created and initialised, we ensure the
+configuration is correct for the builder chosen.
+
+Currently this involves updating the relative link template.
+See ``_update_config_for_builder`` in ``pep_sphinx_extensions/__init__.py``.
+
+
+3.3 Before documents are read
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``create_pep_zero`` hook is called. See `4. PEP 0 generation`_.
+
+
+3.4 Read document
+~~~~~~~~~~~~~~~~~
+
+Parsing the document is handled by ``PEPParser``
+(``pep_sphinx_extensions.pep_processor.parsing.pep_parser.PEPParser``), a light
+wrapper over ``sphinx.parsers.RSTParser``.
+
+``PEPParser`` serves two purposes, firstly to mark the document as containing
+:rfc:`2822` headers, and secondly to register the transforms we want to apply.
+These are:
+
+- ``PEPHeaders``
+- ``PEPTitle``
+- ``PEPContents``
+- ``PEPFooter``
+
+Transforms are then applied in priority order.
+
+
+3.4.1 ``PEPRole`` role
+**********************
+
+This overrides the built-in ``:pep:`` role to return the correct URL.
+
+
+3.4.2 ``PEPHeaders`` transform
+******************************
+
+PEPs start with a set of :rfc:`2822` headers, per :doc:`PEP 1 <../pep-0001>`.
+This transform validates that the required headers are present and of the
+correct data type, and removes headers not for display.
+It must run before the ``PEPTitle`` transform.
+
+
+3.4.3 ``PEPTitle`` transform
+****************************
+
+We generate the title node from the parsed title in the PEP headers, and make
+all nodes in the document children of the new title node.
+This transform must also handle parsing reStructuredText markup within PEP
+titles, such as :doc:`PEP 604 <../pep-0604>`.
+
+
+3.4.4 ``PEPContents`` transform
+*******************************
+
+The automatic table of contents (TOC) is inserted in this transform in a two
+part process.
+
+Firstly the transform inserts a placeholder for the TOC and a horizontal rule
```

Comment:
```suggestion
Firstly, the transform inserts a placeholder for the TOC and a horizontal rule
```

Missing comma

Usage of "firstly", "secondly", is [discouraged](https://www.merriam-webster.com/words-at-play/first-or-firstly) per the opinion of many dictionaries and style guides, and can sound rather pretentious, overcomplicated, distracting or even incorrect to to many readers over the shorter, simpler and more correct "First", "Second", etc,, but its ambiguous enough that I left it for now.

Reply:
Agree on `(ordinal)ly`, thanks for picking it up